docs/DeveloperGuide.md

```diff
@@ -1,38 +1,682 @@
+    <br />
+    <i>Figure 3.4: Class Diagram for DeleteCommand Showing Inheritance of Command</i>
+</p>
+
+The full command for `delete` is `delete [e/ENTRY]`.
+For example, if 'delete' is called, the specific entry inputted in the command is deleted from the list of transactions in
+Moolah Manager.
+
+In a command like `delete 2`:
+
+1. The `main()` method in Duke calls `run()` in Duke. The `ui` reads the command via `ui.readCommand()` and parses it
+   through `CommandParser.parse()`.
+
+2. Within `CommandParser.parse()`, a few functions are called internally.
+   1. `spiltInput()` is called which splits the command from the parameter.
+   2. `getCommand()` is called which searches for the command.
+   3. `ParameterParser.parse()` is called.
+   
+3. Within `ParameterParser.parse()`, a few functions are called internally as well.
+   1. `checkMandatoryTagsExist()` is called where the parameters are checked for all required tags exist based on the command.
+   2. `checkUnsupportedTagsNotExist()` is called to check if the parameter do not contain any unsupported tags based on the command.
+   3. `checkDuplicateTagsNotExist()` is called to check if the parameter do not contain any duplicate tags.
+   4. `checkParameterNotEmpty()` is called to check that the parameter inputted is not empty.
+   5. Once all these checks are successful, `setCommand()` is called.
+   
+4. Within `setCommand()`, more functions are called internally.
+   1. `setParameter()` is called to set the index of the transaction to be deleted.
+   2. The setting is done via `command.setEntryNumber()` which takes in the parameter and executes it in the DeleteCommand Class.
+   3. The parameter, however, needs to be further parsed through the execution of the `parseEntryTag()` function.
+   4. It converts the parameter, which is currently a `String`, to a `Int`.
+   
+5. The delete command is undergoing execution in `command.execute()` which will call functions within the DeleteCommand Class.
+   1. The index, which is the local `entryNumber` variable, goes under further checks by ascertaining whether it is greater than the total
+      number of transactions in the list or lesser than or equal to zero.
+   2. It tells the total size via the local `numberOfTransactions` variable which takes the value called by `transactions.size()`
+      which is located in the TransactionList class.
+   3. Should the above condition be true, it is no longer a valid input and the local `isInputValid` variable is set as false.
+   4. An exception is thrown if `isInputValid` is false. Otherwise, `transactions.deleteTransaction()` is called to remove it.
+   
+6. The above function is called in the TransactionList class which does the following:
+   1. Retrieves the transaction to be deleted via `transactions.get()`.
+   2. Removes it via `transactions.remove()`.
+   
+7. The display shows the successful deletion via `ui.showTransactionAction()` and writes it to file by `storage.writeToFile()`.
```

Comment:
This seems like a very low-level description that's more like a translation of the code into English rather than providing high-level intuition of what's going.